nautobot/ipam/factory.py

```diff
@@ -111,3 +99,85 @@ def prefix(self, n):
             return f"2003:0{n:X}::/32"
 
         raise RuntimeError(f"Don't know how to pick an address for RIR {self.rir.name}")
+
+
+def random_route_distinguisher():
+    """
+    The RD is an arbitrary unique string up to VRF_RD_MAX_LENGTH (21) characters with no explicit validation.
+
+    However by convention it will be one of:
+    - "<2-byte ASN>:<4-byte integer>"
+    - "<IPv4 address>:<2-byte integer>"
+    - "<4-byte ASN>:<2-byte integer>"
+    """
+    branch = randgen.randint(0, 2)
+    if branch == 0:
+        # 16-bit ASNs 64496–64511 are reserved for documentation and sample code
+        return f"{randgen.randint(64496, 64511)}:{randgen.randint(0, 2**32 - 1)}"
+    if branch == 1:
+        # The official stance of factory-boy maintainers is that evaluate() is a private method, but there doesn't
+        # seem to be any convenient alternative way to do this at present.
+        return f"{factory.Faker('ipv4_private').evaluate(None, None, {'locale': None})}:{randgen.randint(0, 2**16 - 1)}"
```

Comment:
For this it would probably be fine to just go straight to Faker
```py
from faker import Faker
>>> Faker().ipv4_private()
'10.100.193.83'
```

Reply:
I need to do some more exploration here - my concern is not having reproducible behavior, if I do this I may need to also be careful to initialize the Faker random seed directly as well as the FactoryBoy random seed?

Reply:
It looks like it works.

```py
>>> factory.random.reseed_random("Nautobot")
>>> factory.Faker("ipv4_private").evaluate(None, None, {"locale": None})
'172.19.147.184'
>>> factory.Faker("ipv4_private").evaluate(None, None, {"locale": None})
'10.87.130.179'
>>> 
>>> factory.random.reseed_random("Nautobot")
>>> Faker().ipv4_private()
'172.19.147.184'
>>> factory.Faker("ipv4_private").evaluate(None, None, {"locale": None})
'10.87.130.179'
```

Reply:
Ah yeah, thanks for testing. Looks like `factory.random.reseed_random()` actually reseeds both libraries :)